test/integration/job/job_test.go

```diff
@@ -171,9 +172,7 @@ func TestJobPodFailurePolicyWithFailedPodDeletedDuringControllerRestart(t *testi
 	restConfig.QPS = 1
 	restConfig.Burst = 1
 	ctx, cancel := startJobControllerAndWaitForCaches(t, restConfig)
-	defer func() {
-		cancel()
-	}()
+	defer cancel()
```

Comment:
use `t.Cleanup`. But also, wrapping it in a function was intentional, because the instance of `cancel` changes throughout the function.

Reply:
Ahh. I just tested this out in go playground. I see the difference now.